examples/regression.jl

```diff
@@ -0,0 +1,178 @@
+# A recreation of https://gpflow.readthedocs.io/en/master/notebooks/advanced/gps_for_big_data.html
+
+using AbstractGPs
+using SparseGPs
+using Distributions
+using LinearAlgebra
+using Optim
+using IterTools
+
+using Plots
+default(; legend=:outertopright, size=(700, 400))
+
+using Random
+Random.seed!(1234)
+
+# %%
+# The data generating function
+function g(x)
+    return sin(3π * x) + 0.3 * cos(9π * x) + 0.5 * sin(7π * x)
+end
+
+N = 10000 # Number of training points
+x = rand(Uniform(-1, 1), N)
+y = g.(x) + 0.3 * randn(N)
+
+scatter(x, y; xlabel="x", ylabel="y", legend=false)
+
+
+# %%
+# A simple Flux model
+using Flux
+
+lik_noise = 0.3
+jitter = 1e-5
+
+struct SVGPModel
+    k # kernel parameters
+    m # variational mean
+    A # variational covariance
+    z # inducing points
+end
+
+@Flux.functor SVGPModel (k, m, A,) # Don't train the inducing inputs
+
+function make_kernel(k)
+    return softplus(k[1]) * (SqExponentialKernel() ∘ ScaleTransform(softplus(k[2])))
+end
+
+# Create the 'model' from the parameters - i.e. return the FiniteGP at inputs x,
+# the FiniteGP at inducing inputs z and the variational posterior over inducing
+# points - q(u).
+function (m::SVGPModel)(x)
+    kernel = make_kernel(m.k)
+    f = GP(kernel)
+    q = MvNormal(m.m, m.A'm.A)
+    fx = f(x, lik_noise)
+    fu = f(m.z, jitter)
```

Comment:
nice - giving the constants proper names really helps making it clearer what's going on!